### PR TITLE
I'm working on making spades logic easier to deal with

### PR DESCRIPTION
**Pull request by davidbrowning**
I also wrote a test for my smallest function to see if I did it right

### CODE REVIEW

**Review by C-Merrill — COMMENTED** [review]
I'd like to see unit tests. The code looks good otherwise, but I won't really know without unit tests

File: test/Test.cpp [hunk 1]
```diff
@@ -82,3 +84,10 @@ BOOST_AUTO_TEST_CASE(SerializeCard)
   BOOST_CHECK_EQUAL(deserializeCard.getSuit(), CLUBS);
   BOOST_CHECK_EQUAL(deserializeCard.getValue(), ACE);
 }
+
+BOOST_AUTO_TEST_CASE(SpadesGetNextPlayer) {
+	Spades s;
+	for (int i = 0; i < 8; i++) {
+		BOOST_CHECK_EQUAL(s.getNextPlayer(i), s.getNextPlayer(i + 1));
+	}
```
Comment (C-Merrill) [inline test/Test.cpp]: I think you've done this right. I'm not sure if those two things are supposed to be equal.

**Comment by mikamerath** [comment]
Before this is merged, wait until we have the Linux fixes. Then fix the merge conflicts

**Comment by C-Merrill** [comment]
You've got merge conflicts now.

**Comment by C-Merrill** [comment]
Fix merge conflicts and request katie to test linux build

**Comment by davidbrowning** [comment]
Fixed the merge conflict

File: source/GameLogic/SpadesLogic.cpp [hunk 2]
```diff
@@ -1,26 +1,24 @@
...
-        void recordMove(std::vector<Card>);
-        Spades(std::vector<Player>);
-        ~Spades() {}
-private:
-        int starter;
-        bool spadesBroken;
-};
-*/
+#include "source\GameLogic\SpadesLogic.hpp"
+
```
Comment (katiesweet) [inline source/GameLogic/SpadesLogic.cpp:2]: Will you please changes these slashes to go the other way? It doesn't build on linux as is.

**Review by katiesweet — CHANGES_REQUESTED** [review]
@davidbrowning will you please change those slashes? They need to be the other way to build on linux. Other than that, it builds just fine. However, your unit tests aren't passing...

screen shot 2016-11-29 at 3 50 19 pm

**Comment by davidbrowning** [comment]
Done! Sorry about that and...yeah that test was only supposed to go to 4. whoops

**Comment by katiesweet** [comment]
Builds on Linux! Thanks for making those changes @davidbrowning!

**Review by C-Merrill — APPROVED** [review]
This builds and looks good. @michaelkamerath good to merge?

**Comment by C-Merrill** [comment]
I ran clang format because it needed it and also to see what changes were actually in the pull request other than format differences.
@davidbrowning remember to run clang format in the future

**Comment by mikamerath** [comment]
@davidbrowning Thanks for doing this, Please comment on what issues this worked with so we can keep track

**Pull request merged**